src/components/Marker.tsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import { ClinicType } from './ClinicFinder';
+
+interface Marker {
+  key: number;
+  id: number;
+  type: ClinicType;
+  lat: number;
+  lng: number;
+  toggleInfoCard: (id: number) => void;
+  activeInfoCardId: number | null;
+  hoveredMarker: number | null;
```

Comment:
optional?

Reply:
no, default state is null. Much more clear imo